### ISSUE TITLE
No way to turn off `-` -> `_` conversion for flag args

### ISSUE THREAD

**Issue by mieubrisse** [argparse4j/argparse4j#126]
I defined an argument to parse like so:
```
    private static final String GOV_CONSTANTS_FILEPATH_ARG = "gov-constants";

        ArgumentParser parser = ArgumentParsers.newFor("Financial Predictor").build()
                .defaultHelp(true)
                .description("A financial modelling CLI");
        parser.addArgument("--" + GOV_CONSTANTS_FILEPATH_ARG)
                .required(true)
                .help("YAML file of gov constants per year");
```
but I was very confused why the following was showing a null value:
```
        String govConstantsFilepath = parsedArgs.getString(GOV_CONSTANTS_FILEPATH_ARG);
        LOG.debug("Gov constants filepath: {}", govConstantsFilepath);
```
When I printed the `Namespace` object, turns out argparse4j is changing the case of the output:
```
Namespace: Namespace(gov_constants=gov-constants.yml)
```
This was very unexpected, and only after some digging did I find the `noDestConversionForPositionalArgs` flag on the builder. However, even with this flag to `true`, my `--gov-constants` flag arg gets registered in the Namespace as `gov_constants`.

It seems likely most people would reuse a single constant for both defining the argument and retrieving the arg from the namespace, so I'm curious what the rationale is for doing the conversion?

**Comment by jstuyts** [comment]
The flag `noDestConversionForPositionalArgs` only applies to positional arguments as is implied by the name. As your argument has a name that starts with a prefix character (assuming you are using the default of a hyphen (`-`)), it is not a positional argument, so the argument name always gets converted.

I understand it is a bit strange to have names that are different from the argument names for the the keys in the resulting namespace. There are 2 reasons for this:

* Emulating the behavior of argparse of Python, which does this so keys are valid attribute names: https://docs.python.org/3/library/argparse.html#dest
* Ensuring the names are valid Java identifiers, so they can also be used to set the arguments on an object using reflection.

What would you prefer?

* A new flag on the configuration that disables conversion of `-` to `_` completely. Setting this flag will also prevent use of the methods that use reflection to fill a user data object.
* Extra methods on `Namespace` that allow retrieving the the values by argument name (without the prefix). For example: `String getStringByArgumentName(String)`.

The second option has the lowest impact, so less chance of introducing backwards incompatibilities or bugs.

**Comment by mieubrisse** [comment]
Hey @jstuyts , thanks for the explanation - makes sense! Two questions:
* Re. option #1, what would the methods that use reflection be? Are there any in argparse4j itself, or just for things like Jackson and so on?
* Re. option number three - would this save both the unprocessed version (e.g. `some-arg`) as well as the processed version, so you can do both?

**Comment by jstuyts** [comment]
@mieubrisse 

* Option 1: Reflection is used when a user data object is passed when passing arguments. For example: [parseArgs(String[], Object)](https://argparse4j.github.io/apidocs/net/sourceforge/argparse4j/inf/ArgumentParser.html#parseArgs-java.lang.String:A-java.lang.Object-)
* Option 3: Yes, it would save both `some-arg` and `some_arg` in the namespace. The new flag would enable this for `parseArgs` methods that return a `Namespace` and that accept a `Map`. I could make it an enumeration so you can choose: underscores (default), argument names, or both.

**Comment by mieubrisse** [comment]
Thanks for clarifying @jstuyts ! With that in mind, I'd also agree that the third option would be best - I don't iterate over the Namespace, so flipping on the flag would allow me to use the same enum everywhere.

**Comment by jstuyts** [comment]
Sorry that it took me a long time to get started. I did end up with a nice naked yak though.

I have added an option to include the names of arguments as keys in the result. This works for positional arguments and named arguments with 1 or 2 dashes as the prefix. It is tested for the main parser, and I will add tests for a sub parser later.

Can you try it out? If you find any issues, let me know, and I will fix them.

**Comment by mieubrisse** [comment]
Haha no worries, I understand how it can go with life getting in the way! Do I just need to upgrade to the latest release to start using this?

**Comment by jstuyts** [comment]
There is no release yet. You will have to build it yourself, or you can download it here: https://1drv.ms/u/s!Ag13WlvRSi9Ytbt2RhRrSpA0ABOfPQ?e=m2uqDA

**Comment by jstuyts** [comment]
@mieubrisse Have you been able to test the changes? I found 1 issue that might affect you after I sent you the build. For named arguments, the flag used for the name in the result could differ from the flag used for dest. This has been corrected in: https://github.com/argparse4j/argparse4j/commit/221a2fb34c3cba77aa83458de4736d1c1dab4568

The artifacts in the link above have been updated.

If it is working for you, I will merge this.
